opal/mca/memory/patcher/configure.m4

```diff
@@ -49,11 +49,15 @@ AC_DEFUN([MCA_opal_memory_patcher_CONFIG],[
 
     AS_IF([test "$opal_memory_patcher_happy" == "yes"], [
         AC_CHECK_FUNCS([__curbrk])
-	AC_CHECK_HEADERS([linux/mman.h sys/syscall.h])
-	AC_CHECK_DECLS([__mmap], [], [], [#include <sys/mman.h>])
-	AC_CHECK_FUNCS([__mmap])
-	AC_CHECK_DECLS([__syscall], [], [], [#include <sys/syscall.h>])
-	AC_CHECK_FUNCS([__syscall])
+    AC_CHECK_HEADERS([linux/mman.h sys/syscall.h])
+# Above sets opal_memory_patcher_happy=no for MacOS/Darwin.
+# Below defines HAVE___mmap, which is only used to support MacOS/Darwin.
+# See issue #6853: on certain systems, mmap will infinitely recurse.
+# if HAVE___mmap is defined.
+#    AC_CHECK_DECLS([__mmap], [], [], [#include <sys/mman.h>])
+#    AC_CHECK_FUNCS([__mmap])
```

Comment:
Per #6853, we know that `__mmap` is defined on Linux systems with a new enough glibc.  If we know for sure that we do not ever need to intercept `__mmap` on Linux, then I'd just delete the two `AC_*` lines here (vs. commenting them out) and amend the comment to explain the situation / history.  Perhaps something like:

> Glibc version X.Y and later have `__mmap`, but the symbol has the same value as `mmap`.  So we only need to intercept one or the other, not both.  We used to intercept `__mmap` on MacOS/Darwin, but per the above logic, this component no longer supports MacOS/Darwin.

Reply:
@jsquyres : Not understanding whether or not  MacOS/Darwin would ever need to intercept __mmap was why we commented out the lines instead of deleting them. Maybe we should leave them commented out for now, then? Or else use the more conservative fix that you suggested in comments for #6853 ?
@hjelmn Thank you for catching my bad indentation changes, Nathan!

Reply:
Just to be clear. We were not intercepting __mmap. On macOS syscall does nothing so for testing the patcher I added the check for __mmap so it could be called from the intercept function. We do not need to support patcher on macOS so the code for __mmap can be safely removed.

Reply:
@hjelmn See https://github.com/open-mpi/ompi/issues/6853#issuecomment-516853917

Reply:
Thank you both for clarifying. I have removed the lines from both configure.m4 and memory_patcher_component.c .